include/fmt/ranges.h

```diff
@@ -591,10 +592,24 @@ struct formatter<join_view<It, Sentinel, Char>, Char> {
     return value_formatter_.parse(ctx);
   }
 
-  template <typename FormatContext>
-  auto format(const join_view<It, Sentinel, Char>& value,
+  template <typename FormatContext, typename Iter,
+            FMT_ENABLE_IF(std::is_copy_constructible<Iter>::value)>
+  auto format(const join_view<Iter, Sentinel, Char>& value,
               FormatContext& ctx) const -> decltype(ctx.out()) {
     auto it = value.begin;
+    return format_impl(value, ctx, it);
+  }
+
+  template <typename FormatContext, typename Iter,
+            FMT_ENABLE_IF(!std::is_copy_constructible<Iter>::value)>
+  auto format(join_view<Iter, Sentinel, Char>& value, FormatContext& ctx) const
+      -> decltype(ctx.out()) {
+    return format_impl(value, ctx, value.begin);
+  }
+
+  template <typename FormatContext>
+  auto format_impl(const join_view<It, Sentinel, Char>& value,
```

Comment:
This should be private. Also let's rename to `do_format` to follow the convention for such methods in {fmt}.